packages/immutadot/src/path/apply.js

```diff
@@ -128,21 +130,22 @@ const apply = operation => {
           return [false, newObj]
         }
 
-        const value = isNil(curObj) ? undefined : curObj[propValue]
```

Comment:
I think it would be better to add another recursive call to `walkPath`, like for slices.
We need `curPath` to be up to date or `pathAlreadyApplied` might cause unexpected mutations with `flow`, I'm going to try to write a failing test.